src/librustc/ty/layout.rs

```diff
@@ -1022,6 +1022,9 @@ pub enum Layout {
         count: u64
     },
 
+    /// The `x86_mmx` type, structs marked with `#[repr(x86_mmx)]`
+    X86Mmx,
```

Comment:
My intuition would be to use `Vector` and special-case it in `rustc_trans` for LLVM.
#45225 changes things a bit, and I'm not sure I'm uncomfortable merging something *before* then, as there would be roughly a dozen individual rebase conflicts, but that's the worst part, figuring out a coherent design is.

All the FFI ABI decisions in #45225 are done without involving the Rust type, intentionally, so probably the best compromise right now would be to have an `x86_mmx` `bool` inside `Vector`.